packages/dynamics-lib/src/entities/base.entity.js

```diff
@@ -263,66 +274,63 @@ export class BaseEntity {
    * @returns {{}} a json representation of the entity state
    */
   toJSON () {
-    // Stringify and parse to recursively call toJSON() on child objects
-    return JSON.parse(JSON.stringify(this._localState))
+    return this._localState
   }
 
   /**
    * Convert the entity into the request body to use in a create/update query to the Dynamics ODATA Web API
    * @returns {{}} the JSON structure to use in a create/update query to the Dynamics ODATA Web API
    */
   toRequestBody () {
-    return Object.entries(this.constructor.definition.mappings)
-      .filter(([, { field }]) => !field.includes('@'))
-      .reduce(
-        (acc, [property, { field }]) => {
-          const serialized = this._toSerialized(property)
-          if (serialized !== undefined) {
-            acc[field] = serialized
-          }
-          return acc
-        },
-        Object.entries(this._bindings).reduce((acc, [k, v]) => {
-          if (v instanceof BaseEntity) {
-            acc[k] = v.id ? `/${v.constructor.definition.dynamicsCollection}(${v.id})` : `$${v.uniqueContentId}`
-          } else {
-            acc[k] = v
-          }
-          return acc
-        }, {})
-      )
+    return Object.entries(this.constructor.definition.mappings).reduce(
+      (acc, [property, { field }]) => {
+        const serialized = this._toSerialized(property)
+        if (serialized !== undefined) {
+          acc[field] = serialized
+        }
+        return acc
+      },
+      Object.entries(this._bindings).reduce((acc, [k, v]) => {
+        if (v instanceof BaseEntity) {
+          acc[k] = v.id ? `/${v.constructor.definition.dynamicsCollection}(${v.id})` : `$${v.uniqueContentId}`
+        } else {
+          acc[k] = v
+        }
+        return acc
+      }, {})
+    )
```

Comment:
This also needs a little more expressiveness to explain what's going on too - it's extremely terse, but quite difficult to work out what's going on (what's k? what's v?), perhaps break out into methods and consider using `map` again rather than reduce as it looks like it might be remapping rather than producing a scalar

Reply:
Yes, agreed.  Some additional comments might help to describe what the method is doing.

Again, cannot use `.map` as we're translating from an array into an object